util/ialloc.c

```diff
@@ -0,0 +1,57 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include <ialloc.h>
+
+int ialloc_alloc(ialloc_t *ia, uint64_t *id)
+{
+    if (ialloc_full(ia)) {
+        return -1;
+    }
+
+    // Give the next free index to the caller
+    *id = ia->head;
+    
+    // Update head to next free index
+    ia->head = ia->idxlist[ia->head];
+
+    // Update number of free indices
+    ia->num_free--;
```

Comment:
I think it's fairly obvious what the code is doing so it's safe to remove these comments.